### PR TITLE
refactor(aws): migrate backup_vault

### PR DESCRIPTION
**Pull request by sinabakh**

### CODE REVIEW

**Comment by sinabakh** [comment]
@aliscott Ready for review.

File: internal/resources/aws/backup_vault.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	Address                       string
+	Region                        string
+	MonthlyEFSWarmBackupGB        *float64 `infracost_usage:"monthly_efs_warm_backup_gb"`
+	MonthlyEFSColdRestoreGB       *float64 `infracost_usage:"monthly_efs_cold_restore_gb"`
+	MonthlyRDSSnapshotGB          *float64 `infracost_usage:"monthly_rds_snapshot_gb"`
+	MonthlyAuroraSnapshotGB       *float64 `infracost_usage:"monthly_aurora_snapshot_gb"`
+	MonthlyDynamodbBackupGB       *float64 `infracost_usage:"monthly_dynamodb_backup_gb"`
+	MonthlyDynamodbRestoreGB      *float64 `infracost_usage:"monthly_dynamodb_restore_gb"`
+	MonthlyFSXWindowsBackupGB     *float64 `infracost_usage:"monthly_fsx_windows_backup_gb"`
+	MonthlyFSXLustreBackupGB      *float64 `infracost_usage:"monthly_fsx_lustre_backup_gb"`
```
Comment (aliscott) [inline internal/resources/aws/backup_vault.go:22]:
```suggestion
	MonthlyFSxWindowsBackupGB     *float64 `infracost_usage:"monthly_fsx_windows_backup_gb"`
	MonthlyFSxLustreBackupGB      *float64 `infracost_usage:"monthly_fsx_lustre_backup_gb"`
```

To be consistent with how we name the `FSx...` resources

File: internal/resources/aws/backup_vault.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	}
+	costComponents = append(costComponents, r.backupVaultCostComponent(bd))
+
+	bd = backupData{ref: "monthly_aurora_snapshot_gb", name: "Aurora snapshot", unit: "GB", usageType: "Aurora:BackupUsage", service: "AmazonRDS", family: "Storage Snapshot", key: "databaseEngine", value: "Aurora PostgreSQL"}
+	if r.MonthlyAuroraSnapshotGB != nil {
+		bd.qty = decimalPtr(decimal.NewFromFloat(*r.MonthlyAuroraSnapshotGB))
+	}
+	costComponents = append(costComponents, r.additionalBackupVaultCostComponent(bd))
+
+	bd = backupData{ref: "monthly_fsx_windows_backup_gb", name: "FSx for windows backup", unit: "GB", usageType: "BackupUsage", service: "AmazonFSx", family: "Storage", key: "fileSystemType", value: "Lustre"}
```
Comment (aliscott) [inline internal/resources/aws/backup_vault.go:98]:
```suggestion
	bd = backupData{ref: "monthly_fsx_windows_backup_gb", name: "FSx for Windows backup", unit: "GB", usageType: "BackupUsage", service: "AmazonFSx", family: "Storage", key: "fileSystemType", value: "Lustre"}
```
Realise this was wrong in the original code

File: internal/resources/aws/backup_vault.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	if r.MonthlyFSXWindowsBackupGB != nil {
+		bd.qty = decimalPtr(decimal.NewFromFloat(*r.MonthlyFSXWindowsBackupGB))
+	}
+	costComponents = append(costComponents, r.additionalBackupVaultCostComponent(bd))
+
+	bd = backupData{ref: "monthly_fsx_lustre_backup_gb", name: "FSx for lustre backup", unit: "GB", usageType: "BackupUsage", service: "AmazonFSx", family: "Storage", key: "fileSystemType", value: "Lustre"}
```
Comment (aliscott) [inline internal/resources/aws/backup_vault.go:104]:
```suggestion
	bd = backupData{ref: "monthly_fsx_lustre_backup_gb", name: "FSx for Lustre backup", unit: "GB", usageType: "BackupUsage", service: "AmazonFSx", family: "Storage", key: "fileSystemType", value: "Lustre"}
```

**Review by aliscott — APPROVED** [review]
LGTM

File: internal/resources/aws/backup_vault.go (hunk 3 above)
Comment (sinabakh) [reply internal/resources/aws/backup_vault.go]: Fixed

File: internal/resources/aws/backup_vault.go (hunk 2 above)
Comment (sinabakh) [reply internal/resources/aws/backup_vault.go]: Fixed

Comment (sinabakh) [reply internal/resources/aws/backup_vault.go]: Fixed

File: internal/resources/aws/backup_vault.go (hunk 3 above)
Comment (sinabakh) [reply internal/resources/aws/backup_vault.go]: Fixed

File: internal/resources/aws/backup_vault.go (hunk 1 above)
Comment (sinabakh) [reply internal/resources/aws/backup_vault.go]: Fixed

**Pull request merged**